src/boosting/gbdt.cpp

```diff
@@ -401,11 +401,16 @@ std::string GBDT::DumpModel() const {
   ss << "\"num_class\":" << num_class_ << "," << std::endl;
   ss << "\"label_index\":" << label_idx_ << "," << std::endl;
   ss << "\"max_feature_idx\":" << max_feature_idx_ << "," << std::endl;
-  if (object_function_ != nullptr) {
-    ss << "\"objective\":\"" << object_function_->GetName() << "\"," << std::endl;
-  }
   ss << "\"sigmoid\":" << sigmoid_ << "," << std::endl;
 
+  // output feature names
+  auto feature_names = std::ref(feature_names_);
+  if (train_data_ != nullptr) {
+    feature_names = std::ref(train_data_->feature_names());
+  }
+
+  ss << "\"feature_names\":" << Common::Join(feature_names.get(), ' ') << std::endl;
```

Comment:
Sorry, still not right here, need to add quotation marks outside string. Should be `ss << "\"feature_names\":\"" << Common::Join(feature_names.get(), ' ') << "\"," << std::endl;
`.

If Common::Join support delimiter as string, `ss << "\"feature_names\":[\"" << Common::Join(feature_names.get(), "\",\"") << "\"]," << std::endl;` will be better.

Could you help reinstall python-package and run simple_example.py to check it?